### PR TITLE
when using findSession check if the session exists in Redis first before creating a new one - a fix for #2019

### PR DESCRIPTION
**Pull request by jchobantonov**

### CODE REVIEW

File: redisson-tomcat/redisson-tomcat-6/src/main/java/org/redisson/tomcat/RedissonSessionManager.java [hunk 1]
```diff
@@ -181,12 +181,20 @@ public Session findSession(String id) throws IOException {
         Session result = super.findSession(id);
         if (result == null) {
             if (id != null) {
-                Map<String, Object> attrs = new HashMap<String, Object>();
+                Map<String, Object> attrs = null;
                 try {
-                    attrs = getMap(id).getAll(RedissonSession.ATTRS);
+                    RMap map = getMap(id);
+                    if (map.isExists()) {
```
Comment (mrniko) [inline redisson-tomcat/redisson-tomcat-6/src/main/java/org/redisson/tomcat/RedissonSessionManager.java]: @jchobantonov 

Why do we need to call for `isExists`  method? if map doesn't exists then attrs object will be empty. So all you need to do is to check if Map empty.

Comment (jchobantonov) [reply redisson-tomcat/redisson-tomcat-6/src/main/java/org/redisson/tomcat/RedissonSessionManager.java]: I know we have system map keys like session:isValid and if the map is empty then it should not exist but I thought there were cases (other bugs) where we could have empty map where those system keys are not set but when I think about it now I think Redis itself do not support empty maps -  so do you want me to change it to what it was originally - checking for empty or if the session:isValid is not true ?

Comment (mrniko) [reply redisson-tomcat/redisson-tomcat-6/src/main/java/org/redisson/tomcat/RedissonSessionManager.java]: > I thought there were cases (other bugs) where we could have empty map where 

We should check such cases then and not adopt the code for them.

> checking for empty or if the session:isValid is not true ?

Check for emptiness

**Comment by jchobantonov** [comment]
@mrniko PR is ready for you

**Comment by mrniko** [comment]
@jchobantonov 

Big thanks!

**Pull request merged**